### PR TITLE
Add config option for chunk folder location

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
Add config option to store chunks in different location

## Motivation and Context
Give admins more power

## How Has This Been Tested?
Local manual test ...

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/dav/lib/Upload/ChunkLocationProvider.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+use OC\Files\Mount\MountPoint;
+use OC\Files\Storage\Local;
+use OCP\Files\Config\IMountProvider;
+use OCP\Files\Storage\IStorageFactory;
+use OCP\IConfig;
+use OCP\IUser;
+
+class ChunkLocationProvider implements IMountProvider {
```
Comment (PVince81) [inline apps/dav/lib/Upload/ChunkLocationProvider.php]: from reading the code it looks like this provider is only for local FS, correct ? (add to PHPDoc)

**Comment by butonic** [comment]
maybe also use it in the  cache file (bad naming here): https://github.com/owncloud/core/blob/master/lib/private/Cache/File.php#L53, used by the legacy filechunking: https://github.com/owncloud/core/blob/master/lib/private/legacy/filechunking.php#L64

**Comment by DeepDiver1975** [comment]
> maybe also use it in the cache file (bad naming here): https://github.com/owncloud/core/blob/master/lib/private/Cache/File.php#L53,

this already exists - see config option 'cache_path' and class CacheMountProvider

this also affects the old chunking

**Review by PVince81 — APPROVED** [review]
:+1: code looks fine

**Comment by DeepDiver1975** [comment]
added PHPDoc and Tests

**Comment by PVince81** [comment]
Merge after CI passed.

@DeepDiver1975 submit stable10 PR ?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 submit stable10 PR ?

done

**Pull request merged**